packages/components/src/FilterBar/context/FilterBarContext.spec.tsx

```diff
@@ -0,0 +1,82 @@
+import React, { useState } from "react"
+import { render, screen, waitFor } from "@testing-library/react"
+import { Filters } from "../types"
+import { FilterBarProvider, useFilterBarContext } from "./FilterBarContext"
+
+type Values = {
+  department: string[]
+  manager: string[]
+  jobTitle: string[]
+}
+
+const MockFilterComponent = ({ id }: { id: string }): JSX.Element => {
+  const { getActiveFilterValues } = useFilterBarContext()
+
+  return <div key={id}>{JSON.stringify(getActiveFilterValues())}</div>
+}
+
+const config = [
+  {
+    id: "department",
+    name: "Department",
+    Component: <MockFilterComponent id="department" />,
+  },
+  {
+    id: "manager",
+    name: "Manager",
+    Component: <MockFilterComponent id="manager" />,
+  },
+  {
+    id: "jobTitle",
+    name: "Job Title",
+    Component: <MockFilterComponent id="jobTitle" />,
+  },
+] satisfies Filters<Values>
+
+export const FilterBarWrapper = ({
+  defaultValues,
+}: {
+  defaultValues: Partial<Values>
+}): JSX.Element => {
+  const [values, setValues] = useState<Partial<Values>>(defaultValues)
+
+  return (
+    <FilterBarProvider
+      filters={config}
+      values={values}
+      onValuesChange={setValues}
+    >
+      {filters => (
+        <>
+          {Object.values(filters).map(({ id, Component }) => (
+            <React.Fragment key={id}>{Component}</React.Fragment>
+          ))}
+        </>
+      )}
+    </FilterBarProvider>
+  )
+}
+
+/*
+ * Note: Traditionally we would like to be testing context specific functionality as part of a greater user flow.
+ *       However, some methods on useFilterBarContext are not yet supported to easily test this. If that's the case,
+ *       we can test them quite explicitly to ensure we have some coverage and confidence.
+ */
+describe("FilterBarContext", () => {
+  describe("useFilterBarContext", () => {
+    it("getActiveFilterValues returns expected values", async () => {
```

Comment:
I know I had previously agreed to making the test here, but I wonder if we could make a simple scenario similar to the user flow you are building within `FilterBar.spec.tsx`. I know the file is long, but it's fine in the meantime until we figure out a convention on how to split it.

- Create 3 single FilterSelects (eg. Department, Manager, Job Title)
- Choose one (Dep) to house attributes of the other two (Source)
- Give one of the sources a default value
- Check the options of the Dep filter has been narrowed according to the Source from above (printing with a test-id might be faster, but otherwise opening the filter to check works)
- Update the value of the other Source
- Check the options of the Dep filter has been further narrowed

Thoughts?

Reply:
Nice. I haven't done exactly this, but I have done up a test that reflects the SR 2.0 use case. :) The test now selects options and does assertions on what arguments are passed to `fetcher` props.